src/ui/public/index_patterns/unsupported_time_patterns.js

```diff
@@ -0,0 +1,34 @@
+import { BoundToConfigObjProvider } from 'ui/bound_to_config_obj';
+
+export function UnsupportedTimePatternsProvider(Private, $injector) {
+  const BoundToConfigObj = Private(BoundToConfigObjProvider);
+  const sessionStorage = $injector.get('sessionStorage');
+  const kbnUrl = $injector.get('kbnUrl');
+  const Notifier = $injector.get('Notifier');
+
+  const notify = new Notifier({ location: 'Index Patterns' });
+
+  const HISTORY_STORAGE_KEY = 'indexPatterns:warnAboutUnsupportedTimePatterns:history';
+  const FLAGS = new BoundToConfigObj({
+    enabled: '=indexPatterns:warnAboutUnsupportedTimePatterns'
+  });
+
+  return function warnAboutUnsupportedTimePattern(indexPattern) {
+    if (!FLAGS.enabled) {
+      return;
+    }
+
+    const previousIds = sessionStorage.get(HISTORY_STORAGE_KEY) || [];
+    if (previousIds.includes(indexPattern.id)) {
+      return;
+    }
+
+    sessionStorage.set(HISTORY_STORAGE_KEY, [...previousIds, indexPattern.id]);
+    notify.warning(
```

Comment:
I definitely agree about separating concerns, but `indexPattern.isUnsupportedTimePattern()` [already exists](https://github.com/spalger/kibana/blob/a437d623c272722a11623a07fe25d17a843ef983/src/ui/public/index_patterns/_index_pattern.js#L332-L334) and is responsible for determining if [this logic should execute](https://github.com/spalger/kibana/blob/a437d623c272722a11623a07fe25d17a843ef983/src/ui/public/index_patterns/_index_pattern.js#L95-L97) as well as [if we should show the warning on the edit page](https://github.com/spalger/kibana/blob/a437d623c272722a11623a07fe25d17a843ef983/src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/edit_index_pattern.html#L35).

This module used to be implemented [in the caller](https://github.com/elastic/kibana/pull/12158/files#diff-274005236556ec7aa3ebc5475ca54f5bR96), and the caller was responsible for determining if the unsupported index pattern should trigger a warning, but in an effort to make it more testable I pulled it out into a module that is purely responsible for determining if we should call `notify.warning()` or not.

All that said, I'm not fond of how much stubbing is necessary to test this module and would be interested in other ways you think this could be broken up to make that less necessary. At the end of the day though, I'm pretty sure we are going to have to stub the browser storage if we're going to test how this or any other module interacts with session storage, and we're going to have to stub notify for the same reason... Open to other suggestions though.